packages/rum/src/domain/record/shadowDom.ts

```diff
@@ -0,0 +1,74 @@
+import { addTelemetryDebug, DOM_EVENT } from '@datadog/browser-core'
+import type { RumConfiguration } from '@datadog/browser-rum-core'
+import { startMutationObserver } from './mutationObserver'
+import { initInputObserver } from './observers'
+import type { MutationCallBack, InputCallback } from './observers'
+
+interface ShadowDomCallBacks {
+  stop: () => void
+  flush: () => void
+}
+
+export type ShadowDomCallBack = (shadowRoot: ShadowRoot) => void
+
+export interface ShadowRootsController {
+  addShadowRoot: ShadowDomCallBack
+  removeShadowRoot: ShadowDomCallBack
+  stop: () => void
+  flush: () => void
+}
+
+export const initShadowRootsController = (
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏what about renaming `shadowDom.ts` to `shadowRootsController.ts`?